app/routers/file_streamer.py

```
@@ -0,0 +1,46 @@
+
+    if gcs_file_path not in available_files:
+        print(f"❌ File {gcs_file_path} NOT FOUND in GCS")
+        raise HTTPException(status_code=404, detail="File not found in GCS")
+
+    blob = bucket.blob(gcs_file_path)
+    try:
+        pdf_bytes = blob.download_as_bytes()
+    except Exception:
+        # Perbaiki logging agar tidak error
+        logging.error("Traceback: %s", traceback.format_exc())
+        raise HTTPException(status_code=500, detail="Failed to download PDF from GCS")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)